broker.go

```diff
@@ -773,8 +773,8 @@ func (b *Broker) Producer(conf ProducerConf) Producer {
 //
 // Upon a successful call, the message's Offset field is updated.
 func (p *producer) Produce(topic string, partition int32, messages ...*proto.Message) (offset int64, err error) {
-
-	for retry := 0; retry < p.conf.RetryLimit; retry++ {
+	retry := 0
```

Comment:
why not to leave `retry` as a loop counter and move check from line 807 to err check ` case nil` at line 785?

Reply:
Agree with @hashmap

Reply:
Let's remove this declaration, the original version of the loop should work now